python/src/py_opcua_module.cpp

```diff
@@ -242,10 +242,16 @@ static void Node_SetValue(Node & self, const object & obj, VariantType vtype)
 // UaClient helpers
 //--------------------------------------------------------------------------
 
+template<typename T>
+boost::shared_ptr<T> make_shared_ptr(std::shared_ptr<T>& ptr)
+{
+    return boost::shared_ptr<T>(ptr.get(), [ptr](T*) mutable {ptr.reset();});
+}
```

Comment:
make_shared_ptr is wrapping a std::shared_ptr in a boost::shared_ptr by holding a reference to the std::shared_ptr in the custom deleter callback of the boost::shared_ptr.

To replace all boost::shared_ptr from the python wrapper could be a good idea, but as I don't use this wrapper I would be the wrong one to do this - I would simply not recognize errors in this interface (at least no errors which are not part of the test suite).